app/integration_action.go

```diff
@@ -207,6 +207,23 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 // Perform an HTTP POST request to an integration's action endpoint.
 // Caller must consume and close returned http.Response as necessary.
 func (a *App) DoActionRequest(rawURL string, body []byte) (*http.Response, *model.AppError) {
+	inURL, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, model.NewAppError("DoActionRequest", "api.post.do_action.action_integration.app_error", nil, err.Error(), http.StatusBadRequest)
+	}
+
+	siteURL, _ := url.Parse(*a.Config().ServiceSettings.SiteURL)
+	rawURLPath := path.Clean(rawURL)
+	if siteURL != nil && (strings.HasPrefix(rawURLPath, "/plugins/") || strings.HasPrefix(rawURLPath, "plugins/")) {
+		if siteURL.Path == "" {
+			siteURL.Path = "/"
```

Comment:
Instead of mutating `siteURL.Path` here, I propose just adding `/` to the join below:
```
		inURL.Path = path.Join("/", siteURL.Path, rawURLPath)
```

Reply:
I agree, much cleaner solution.

Reply:
The only problem, and the main reason I wrote it that way is that a check on inURL.Path that comes a few lines later is going to fail in the case the input ```rawURL``` doesn't have a leading slash.

Reply:
Hmm, perhaps I misunderstand -- with my proposal, `inURL.Path` will always have a leading slash if it enters this block. (If it doesn't, it wouldn't prior to my proposal either.) Or am I misunderstanding the concern?

Reply:
I think my brain short circuited for a moment there. That change works fine.

Reply:
Sounds good -- disregard my comment below, as you correctly point out it's only used if it's a valid plugin path. I think if we just patch up this `path.Join`, we're good!